src/functions/genfrac.js

```diff
@@ -321,3 +335,137 @@ defineFunction({
         }, parser.mode);
     },
 });
+
+const stylArray = ["display", "text", "script", "scriptscript"];
+
+const delimFromValue = function(delimString: string): string | null {
+    let delim = null;
+    if (delimString.length > 0) {
+        delim = delimString;
+        delim = delim === "." ? null : delim;
+    }
+    return delim;
+};
+
+defineFunction({
+    type: "genfrac",
+    names: ["\\genfrac"],
+    props: {
+        numArgs: 6,
+        greediness: 6,
+        argTypes: ["math", "math", "size", "text", "math", "math"],
+    },
+    handler: ({parser}, args) => {
+        const numer = args[4];
+        const denom = args[5];
+
+        // Look into the parse nodes to get the desired delimiters.
+        let leftNode = checkNodeType(args[0], "ordgroup");
+        if (leftNode) {
+            leftNode = assertNodeType(leftNode.value[0], "open");
+        } else {
+            leftNode = assertNodeType(args[0], "open");
+        }
+        const leftDelim = delimFromValue(leftNode.value);
+
+        let rightNode = checkNodeType(args[1], "ordgroup");
+        if (rightNode) {
+            rightNode = assertNodeType(rightNode.value[0], "close");
+        } else {
+            rightNode = assertNodeType(args[1], "close");
+        }
+        const rightDelim = delimFromValue(rightNode.value);
+
+        const barNode = assertNodeType(args[2], "size");
+        let hasBarLine;
+        let barSize = null;
+        if (barNode.value.isBlank) {
+            // \genfrac acts differently than \above.
+            // \genfrac treats an empty size group as a signal to use a
+            // standard bar size. \above would see size = 0 and omit the bar.
+            hasBarLine = true;
+        } else {
+            barSize = barNode.value.value;
+            hasBarLine = barSize.number > 0;
+        }
+
+        // Find out if we want displaystyle, textstyle, etc.
+        let size = "auto";
+        let styl = checkNodeType(args[3], "ordgroup");
+        if (styl) {
+            if (styl.value.length > 0) {
+                size = stylArray[Number(styl.value[0].value)];
+            }
+        } else {
+            styl = assertNodeType(args[3], "textord");
+            size = stylArray[Number(styl.value)];
```

Comment:
Is this really how TeX does this?   I guess `\genfrac` is pretty low-level but I would've expected some sort of identifier or something.

Reply:
`\genfrac` is an AMS function, so I'll quote from the *User’s Guide for the amsmath Package*, 

>the  fourth  argument is a mathstyle override:  integer values 0–3 select respectively `\displaystyle`, `\textstyle`, `\scriptstyle`, and `\scriptscriptstyle`.

The *Guide* goes on to suggest that user's should use `\genfrac` to define their own macro with a short name, rather than repeatedly use `\genfrac` in full.